RequestClientCert option can cause connection error

### Your environment.
  - Version: b28063ed43f57a8c2d54e36327e8ae6c3d40acf8
  - Browser: N/A
  - Other Information: OpenSSL 1.1.1c

### What did you do?

Server: `go run example_server.go`
```
package main
  
import (
        "fmt"
        "net"
        "time"

        "github.com/pion/dtls"
        "github.com/pion/dtls/examples/util"
)

func main() {
    addr := &net.UDPAddr{IP: net.ParseIP("127.0.0.1"), Port: 4444}

    certificate, privateKey, genErr := dtls.GenerateSelfSigned()
    util.Check(genErr)

    config := &dtls.Config{
        Certificate:          certificate,
        PrivateKey:           privateKey,
        ClientAuth:           dtls.RequestClientCert,
    }

    listener, err := dtls.Listen("udp", addr, config)
    util.Check(err)
    defer func() {
        util.Check(listener.Close())
    }()

    fmt.Println("Listening")

    _, err := listener.Accept()
    util.Check(err)
}
```

Client: `openssl s_client -dtls1_2 -connect 127.0.0.1:4444 -debug`

### What did you expect?
The connection to succeed without error

### What happened?
```
Listening
panic: dtls: buffer is too small
```

--- a/handshake_message_certificate.go
+++ b/handshake_message_certificate.go
@@ -26,12 +26,15 @@ func (h *handshakeMessageCertificate) Marshal() ([]byte, error) {
 }
 
 func (h *handshakeMessageCertificate) Unmarshal(data []byte) error {
-	if len(data) < 6 {
+	if len(data) < 3 {
 		return errBufferTooSmall
 	}
 
 	certificateBodyLen := int(bigEndianUint24(data))
 	certificateLen := int(bigEndianUint24(data[3:]))
+	if certificateLen == 0 {
+		return nil
+	}
 	if certificateBodyLen+3 != len(data) {
 		return errLengthMismatch
 	} else if certificateLen+6 != len(data) {

--- a/handshake_message_certificate_test.go
+++ b/handshake_message_certificate_test.go
@@ -74,3 +74,22 @@ func TestHandshakeMessageCertificate(t *testing.T) {
 		t.Errorf("handshakeMessageCertificate marshal: got %#v, want %#v", raw, rawCertificate)
 	}
 }
+
+func TestEmptyHandshakeMessageCertificate(t *testing.T) {
+	rawCertificate := []byte{
+		0x00, 0x00, 0x00,
+	}
+
+	expectedCertificate := &handshakeMessageCertificate{
+		certificate: nil,
+	}
+
+	c := &handshakeMessageCertificate{}
+	if err := c.Unmarshal(rawCertificate); err != nil {
+		t.Error(err)
+	}
+
+	if !reflect.DeepEqual(c, expectedCertificate) {
+		t.Errorf("handshakeMessageCertificate unmarshal: got %#v, want %#v", c, expectedCertificate)
+	}
+}

